unified_planning/model/contingent/environment.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2023 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import random
+import unified_planning as up
+from unified_planning.exceptions import UPUsageError
+from pysmt.shortcuts import Solver, Not, And, Symbol, Or, ExactlyOne, EqualsOrIff
+from pysmt.oracles import get_logic
+from typing import Dict
+
+
+class Environment:
```

Comment:
Is `Environment` the right name? It might create conflicts with the `up.environment.Environment` class